src/components/_badge.scss

```diff
@@ -21,62 +21,64 @@ $badge-padding-x: (
   color: map-get(variables.$colors, "heading");
   padding: map-get($badge-padding-y, "sm") map-get($badge-padding-x, "lg");
   border-radius: map-get(variables.$radius, "xs");
-}
 
-.badge.small {
-  padding: 0.5px 5px;
-  font-size: 0.7rem;
-}
+  &.small {
+    padding: 0.5px 5px;
+    font-size: 0.7rem;
+  }
 
-.badge.large {
-  padding: map-get($badge-padding-y, "lg") map-get($badge-padding-x, "xl");
-  border-radius: map-get(variables.$radius, "lg");
-}
+  &.large {
+    padding: map-get($badge-padding-y, "lg") map-get($badge-padding-x, "xl");
+    border-radius: map-get(variables.$radius, "lg");
 
-.badge .icon {
-  font-size: 1.1rem;
-}
+    .icon {
+      font-size: 1.3rem;
+    }
+  }
 
-.badge.large .icon {
-  font-size: 1.3rem;
-}
+  .icon {
+    font-size: 1.1rem;
+  }
```

Comment:
First place everything for component's base case, without modifiers, then everything else comes